src-java/kilda-model/src/main/java/org/openkilda/model/Cookie.java

```diff
@@ -44,102 +38,133 @@
+            ServiceCookie.ServiceCookieTag.MULTITABLE_TRANSIT_DROP_COOKIE).getValue();
+    public static final long LLDP_INPUT_PRE_DROP_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.LLDP_INPUT_PRE_DROP_COOKIE).getValue();
+    public static final long LLDP_TRANSIT_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.LLDP_TRANSIT_COOKIE).getValue();
+    public static final long LLDP_INGRESS_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.LLDP_INGRESS_COOKIE).getValue();
+    public static final long LLDP_POST_INGRESS_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.LLDP_POST_INGRESS_COOKIE).getValue();
+    public static final long LLDP_POST_INGRESS_VXLAN_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.LLDP_POST_INGRESS_VXLAN_COOKIE).getValue();
+    public static final long LLDP_POST_INGRESS_ONE_SWITCH_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.LLDP_POST_INGRESS_ONE_SWITCH_COOKIE).getValue();
+    public static final long ARP_INPUT_PRE_DROP_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_INPUT_PRE_DROP_COOKIE).getValue();
+    public static final long ARP_TRANSIT_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_TRANSIT_COOKIE).getValue();
+    public static final long ARP_INGRESS_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_INGRESS_COOKIE).getValue();
+    public static final long ARP_POST_INGRESS_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_POST_INGRESS_COOKIE).getValue();
+    public static final long ARP_POST_INGRESS_VXLAN_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_POST_INGRESS_VXLAN_COOKIE).getValue();
+    public static final long ARP_POST_INGRESS_ONE_SWITCH_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_POST_INGRESS_ONE_SWITCH_COOKIE).getValue();
 
-    // 9 bits cookie type "field"
-    public static final long TYPE_MASK                               = 0x1FF0_0000_0000_0000L;
-    public static final long FLOW_COOKIE_TYPE                        = 0x0000_0000_0000_0000L;
-    public static final long LLDP_INPUT_CUSTOMER_TYPE                = 0x0010_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VLAN_EGRESS_RULES_TYPE   = 0x0020_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VXLAN_EGRESS_RULES_TYPE  = 0x0030_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VXLAN_TRANSIT_RULES_TYPE = 0x0040_0000_0000_0000L;
-    public static final long MULTITABLE_INGRESS_RULES_TYPE           = 0x0050_0000_0000_0000L;
-    public static final long ARP_INPUT_CUSTOMER_TYPE                 = 0x0060_0000_0000_0000L;
+    @JsonCreator
+    public Cookie(long value) {
+        super(value);
+    }
 
-    private final long value;
+    @Builder
+    public Cookie(CookieType type) {
```

Comment:
You can allow the builder to set both value and type, but as for now use only the type.

Reply:
Em... why for? This 2 different use cases - one "restore" cookie from the raw value (direct `new Cookie(raw)` call) and another build new cookie from fields provided by the app (builder way). I do not want to add `value` field into the builder, to will make it less obvious.

Reply:
Got it.